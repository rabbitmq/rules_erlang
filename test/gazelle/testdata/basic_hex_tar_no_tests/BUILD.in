# gazelle:erlang_no_tests
# gazelle:erlang_generate_fewer_bytecode_rules
