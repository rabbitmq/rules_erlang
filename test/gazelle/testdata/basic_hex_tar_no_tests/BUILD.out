load("@rules_erlang//:erlang_bytecode2.bzl", "erlang_bytecode", "erlc_opts")
load("@rules_erlang//:erlang_app.bzl", "erlang_app")
load("@rules_erlang//:untar.bzl", "untar")

# gazelle:erlang_no_tests
# gazelle:erlang_generate_fewer_bytecode_rules

untar(
    name = "contents",
    outs = [
        "LICENSE",
        "LICENSE-APACHE2",
        "LICENSE-MPL-RabbitMQ",
        "README.md",
        "rebar.config",
        "rebar.lock",
        "src/aten.app.src",
        "src/aten.erl",
        "src/aten_app.erl",
        "src/aten_detect.erl",
        "src/aten_detector.erl",
        "src/aten_emitter.erl",
        "src/aten_sink.erl",
        "src/aten_sup.erl",
    ],
    archive = "contents.tar.gz",
)

erlc_opts(
    name = "erlc_opts",
    values = select({
        "@rules_erlang//:debug_build": [
            "+debug_info",
        ],
        "//conditions:default": [
            "+debug_info",
            "+deterministic",
        ],
    }),
    visibility = [":__subpackages__"],
)

erlang_bytecode(
    name = "other_beam",
    srcs = [
        "src/aten.erl",
        "src/aten_app.erl",
        "src/aten_detect.erl",
        "src/aten_detector.erl",
        "src/aten_emitter.erl",
        "src/aten_sink.erl",
        "src/aten_sup.erl",
    ] + glob(["src/**/*.erl"]),
    hdrs = [":public_and_private_hdrs"],
    app_name = "aten",
    dest = "ebin",
    erlc_opts = "//:erlc_opts",
)

filegroup(
    name = "beam_files",
    srcs = [":other_beam"],
)

filegroup(
    name = "srcs",
    srcs = [
        "src/aten.erl",
        "src/aten_app.erl",
        "src/aten_detect.erl",
        "src/aten_detector.erl",
        "src/aten_emitter.erl",
        "src/aten_sink.erl",
        "src/aten_sup.erl",
    ] + glob([
        "src/**/*.app.src",
        "src/**/*.erl",
    ]),
)

filegroup(
    name = "private_hdrs",
    srcs = glob(["src/**/*.hrl"]),
)

filegroup(
    name = "public_hdrs",
    srcs = glob(["include/**/*.hrl"]),
)

filegroup(
    name = "priv",
    srcs = glob(["priv/**/*"]),
)

filegroup(
    name = "license_files",
    srcs = [
        "LICENSE",
        "LICENSE-APACHE2",
        "LICENSE-MPL-RabbitMQ",
    ] + glob(["LICENSE*"]),
)

filegroup(
    name = "public_and_private_hdrs",
    srcs = [
        ":private_hdrs",
        ":public_hdrs",
    ],
)

filegroup(
    name = "all_srcs",
    srcs = [
        ":public_and_private_hdrs",
        ":srcs",
    ],
)

erlang_app(
    name = "erlang_app",
    srcs = [":all_srcs"],
    hdrs = [":public_hdrs"],
    app_description = "Erlang node failure detector",
    app_name = "aten",
    app_version = "0.5.8",
    beam_files = [":beam_files"],
    extra_apps = [
        "crypto",
        "sasl",
    ],
    license_files = [":license_files"],
    priv = [":priv"],
)

alias(
    name = "aten",
    actual = ":erlang_app",
    visibility = ["//visibility:public"],
)
